services/thermal_mgr/thermal_mgr.c

```diff
@@ -44,17 +45,51 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
 
+  if (event == NULL) {
+    return ERR_CODE_INVALID_ARG;
+  }
+
+  if (xQueueSend(thermalMgrQueueHandle, event, 0) != pdTRUE) {
+    return ERR_CODE_QUEUE_FULL;
+  }
+
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+  // Send an event to handle the over-temperature and exit the interrupt ASAP
+  thermal_mgr_event_t event;
+  event.type = THERMAL_MGR_EVENT_OS_INTERRUPT;
+  thermalMgrSendEvent(&event);
 }
 
 static void thermalMgr(void *pvParameters) {
-  /* Implement this task */
   while (1) {
-    
+    thermal_mgr_event_t event;
+
+    if (xQueueReceive(thermalMgrQueueHandle, &event, 0) == pdTRUE) {
+      // An event was received
+      error_code_t errCode;
+      float temp = 0.0f;
+
+      LOG_IF_ERROR_CODE(readTempLM75BD(LM75BD_OBC_I2C_ADDR, &temp));
+
+      if (event.type == THERMAL_MGR_EVENT_MEASURE_TEMP_CMD) {
+        addTemperatureTelemetry(temp);
+      } else if (event.type == THERMAL_MGR_EVENT_OS_INTERRUPT) {
+        // Determine if it's over-temperature or hysteresis, and act accordingly
+        if (temp >= LM75BD_DEFAULT_OT_THRESH) {
+          overTemperatureDetected();
+        } else if (temp <= LM75BD_DEFAULT_HYST_THRESH) {
+          safeOperatingConditions();
+        } else {
+          LOG_ERROR_CODE(ERR_CODE_INVALID_STATE);
+        }
+      } else {
+        // The event type is unknown
+        LOG_ERROR_CODE(ERR_CODE_INVALID_STATE);
```

Comment:
Probably beat to define a new error code for invalid events such as ERR_CODE_INVALID_EVENT